integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,503 @@
+        .until(runJmsClient(new String(javasCmd)));
+  }
+
+  // Download the wlthint3client.jar from Adminserver pod to local filesystem.
+  // Use wlthint3client.jar in classpath to build and run the standalone 
+  // JMS client that sends messages to a Uniform Distributed Queue using 
+  // load balancer http(s) url which maps to custom channel on cluster member
+  // server on WebLogic cluster.
+  public void buildClient() {
+
+    StringBuffer copyCmd = new StringBuffer("kubectl cp ");
+    copyCmd.append(domainNamespace + "/" + adminServerPodName);
+    copyCmd.append(":/u01/oracle/wlserver/server/lib/wlthint3client.jar ");
+    copyCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar"));
+    logger.info("kubectl copy command {0}", copyCmd.toString());
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(copyCmd), true));
+    logger.info("kubectl copy returned {0}", result.toString());
+
+    StringBuffer javacCmd = new StringBuffer("javac -cp ");
+    javacCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar "));
+    javacCmd.append(Paths.get(RESOURCE_DIR, "tunneling", "JmsTestClient.java"));
+    javacCmd.append(Paths.get(" -d "));
+    javacCmd.append(Paths.get(RESULTS_ROOT));
+    logger.info("javac command {0}", javacCmd.toString());
+    result = assertDoesNotThrow(
+        () -> exec(new String(javacCmd), true));
+    logger.info("javac returned {0}", result.toString());
+    logger.info("javac returned EXIT value {0}", result.exitValue());
+    assertTrue(result.exitValue() == 0, "Client compilation fails");
+  }
+
+  // Run external standalone JMS Client using wlthint3client.jar in classpath.
+  // The client sends 300 messsage to a Uniform Distributed Queue.
+  // Make sure that each destination get excatly 150 messages each.
+  private static Callable<Boolean> runJmsClient(String javaCmd) {
+    return (()  -> {
+      ExecResult result = assertDoesNotThrow(() -> exec(new String(javaCmd), true));
+      logger.info("java returned {0}", result.toString());
+      logger.info("java returned EXIT value {0}", result.exitValue());
+      return ((result.exitValue() == 0));
+    });
+  }
+
+  @AfterAll
+  void tearDown() {
+
+    StringBuffer deployIngress = new StringBuffer("kubectl delete -f ");
+    deployIngress.append(Paths.get(RESULTS_ROOT, "voyager.tunneling.yaml"));
+    assertDoesNotThrow(() -> exec(new String(deployIngress), true));
+
+    StringBuffer deployTlsIngress = new StringBuffer("kubectl delete -f ");
+    deployTlsIngress.append(Paths.get(RESULTS_ROOT, "voyager.tls.tunneling.yaml"));
+    assertDoesNotThrow(() -> exec(new String(deployTlsIngress), true));
+
+    // uninstall Voyager
+    if (voyagerHelmParams != null) {
+      assertThat(uninstallVoyager(voyagerHelmParams))
+          .as("Test uninstallVoyager returns true")
+          .withFailMessage("uninstallVoyager() did not return true")
+          .isTrue();
+    }
+  }
+
+  private static void createCertKeyFiles(String cn) {
```

Comment:
Add some comments as to what these private methods does.

Reply:
modified the scope to private and added description